crates/miden-lib/src/account/faucets/mod.rs

```diff
@@ -197,10 +197,26 @@ pub fn create_basic_fungible_faucet(
     account_storage_mode: AccountStorageMode,
     auth_scheme: AuthScheme,
 ) -> Result<(Account, Word), FungibleFaucetError> {
-    // Atm we only have RpoFalcon512 as authentication scheme and this is also the default in the
-    // faucet contract.
-    let auth_component: RpoFalcon512 = match auth_scheme {
-        AuthScheme::RpoFalcon512 { pub_key } => RpoFalcon512::new(pub_key),
+    // TODO in actual implementation, we should probably get the root of the distribute procedure
+    // in a more reliable way.
+    let mut distribute_proc_root = None;
+    for module in basic_fungible_faucet_library().module_infos() {
+        for (_, procedure_info) in module.procedures() {
+            if procedure_info.name.contains("distribute") {
+                distribute_proc_root = Some(procedure_info.digest);
+                break;
+            }
+        }
+        if distribute_proc_root.is_some() {
+            break;
+        }
+    }
+    let distribute_proc_root = distribute_proc_root.expect("distribute procedure should exist");
```

Comment:
Ideally, this would be done as something like:
```rust
let distribute_proc_root = BasicFungibleFaucet::procedures().distribute;
```
Where `BasicFungibleFaucet::procedures()` returns something like:
```rust
pub struct BasicFungibleFaucetProcedures {
    pub distribute: Digest,
    pub burn: Digest,
}
```
Internally, we'd probably use something like the loop you have here to build `BasicFungibleFaucetProcedures` (in a lazy-static way).

We could also try to use constants like:
```rust
let distribute_proc_root = BasicFungibleFaucet::DISTRIBUTE_PROC_ROOT;
```
But these would probably need to be populated as build-time somehow, and that's more trouble than it's worth.

Reply:
@Fumuran also did something like the above in #1532 (see https://github.com/0xMiden/miden-base/pull/1532/files#r2186286970).